test/e2e/dns_ingressdegrade_test.go

```diff
@@ -0,0 +1,102 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorclient "github.com/openshift/cluster-ingress-operator/pkg/operator/client"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TestIngressStatus - degrade/restore status via DNS config
+// This test will check the ingress status
+//
+// Steps :-
+// 1. initial check - should be false
+// 2. update the DNS private zone tags to unknown value
+// 3. check the status should have degraded to true
+// 4. re-instate the original condition
+// 5. check the status should have degraded to false
+func TestIngressStatus(t *testing.T) {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		fmt.Printf("failed to get kube config: %s\n", err)
+		os.Exit(1)
```

Comment:
Use `t.Fatalf()` here instead of exiting the suite. Let the testing suite clean up.